### DISCUSSION TITLE
Exception handling job constructor

### DISCUSSION THREAD

**Discussion by KoalaBear84** [quartznet/quartznet#2391] [Q&A]
I was wondering for weeks why some job only ran once. Just found out it was because of an exception in a constructor.

Had this issue before, trying to set a `BaseAddress` multiple times is not something the `HttpClient` likes 😅

Now I would like to get some sort of error logging when this happens again in the future.

Are there any ways to catch exceptions in IJob constructors? Tried to look for something, but didn't find any documentation or examples of this.

Thanks!

**Comment by lahma** [comment]
Generally Quartz should log this with an error: https://github.com/quartznet/quartznet/blob/3.x/src/Quartz/Simpl/SimpleJobFactory.cs . I think it was almost 20 years ago when a co-worker was investigating why Job was not running and it was a problem in the constructor, throwing an exception.

**Reply by KoalaBear84** [reply]
I guess you are right 😅 I dug a hole and fell in myself.. And it is quite related to the other thread I 'just' replied to (https://github.com/quartznet/quartznet/discussions/2372) 🙃

```xml
  <rules>
    <logger final="false" name="*Quartz*" minlevel="Warning" writeTo="fileLogger" />
    <logger final="true" name="*Quartz*" minlevel="Trace" writeTo="quartzSchedulerLogging" />
    <logger name="*" minlevel="Trace" writeTo="fileLogger" />
  </rules>
```

I was logging all the Quartz logging to another file 🥳 #OMG

So now added the first line so all of Warning and above (Error) ALSO goes to the file.

Thanks for the reply! Got me checking everything again 👍
